lib/etl/transform/transform_json.rb

```diff
@@ -0,0 +1,38 @@
+require 'json'
+require 'json-schema'
+
+class TransformJson
+  def process(input_json)
+    @input_json = input_json
+    pl = parsed_list
+    JSON::Validator.validate(schema_fname, pl)
+    pl
+    
+  rescue => error
+    ap "Error opening json file: #{error}"
+  end
+
+  private
+
+  attr_reader :input_json, :schema_fname
+  
+  def schema_fname
+    raise NotImplementedError.new("#{self.class.name}#area is an abstract method.")
+  end
+  
+  def parsed_list
+    JSON.parse(File.read(input_json), :symbolize_names => true)
+  end
+end
+
+class TransformJsonResources < TransformJson
```

Comment:
No.  The JSON schema specifies an array, and hence the processing of the JSON acts on an array of data.

The reason for an array is that in general most of our data is lots of small records.  It would be extremely awkward/ugly to require having a JSON file for each record (of resources, groups, etc.).  As implemented, you can have as many records in a JSON file as you want.  For things like books, having 1 is natural, for things like articles (and our current group data) having a much larger number of resources per JSON file is more natural.